packages/core/app-layout/src/components/navbar/AccountDropdown.vue

```diff
@@ -0,0 +1,69 @@
+<template>
+  <KDropdownMenu
+    class="account-dropdown"
+    :kpop-attributes="{ placement: 'bottomStart' }"
+    :label="userInitials"
+    size="small"
+    :width="width"
+  >
+    <template #items="{ handleSelection }">
+      <slot>
+        <KDropdownItem
+          v-for="option in options"
+          :key="option.value"
+          :class="{ 'has-divider': option.hasDivider }"
+          :data-testid="'select-' + option.value"
+          :has-divider="option.hasDivider"
+          @click="handleSelection(option)"
+        >
+          <slot :option="option">
+            {{ option.label }}
+          </slot>
+        </KDropdownItem>
+      </slot>
+    </template>
+  </KDropdownMenu>
+</template>
+
+<script setup lang="ts">
+import { PropType } from 'vue'
+import type { DropdownItem } from '@kong/kongponents'
+
+defineProps({
+  options: {
+    type: Array as PropType<DropdownItem[]>,
+    required: true,
+  },
+  width: {
+    type: String,
+    default: '240',
+  },
+  userInitials: {
+    type: String,
+    required: true,
+  },
```

Comment:
If for some reason the host app doesn't pass in the user's initials (let's pretend somehow their name isn't set in session data) the entire dropdown will fail to render.

If we default the prop to a `string` with a single space character, the dropdown will at least render.

![image](https://user-images.githubusercontent.com/2229946/226704547-01d94b0f-99b3-4a74-b90f-113b86f39a88.png)

```suggestion
  userInitials: {
    type: String,
    required: true,
    // Important: default to a string with a single empty space to force the dropdown to always be visible
    default: ' ',
  },
```

Reply:
Interesting, added. 
I thought we wouldn't want to render the dropdown at all in that case, but I see this is a better approach.

Reply:
I'm probably over-optimizing; I'm thinking of a state where the session data APIs fail and the user gets stuck to where they still need to be able to click the logout link.